File: feature/wanandroid/pkg/src/main/java/com/open9527/wanandroid/pkg/net/share/ShareApi.java
package com.open9527.wanandroid.pkg.net.share;

import com.android.open9527.okhttp.annotation.HttpIgnore;
import com.android.open9527.okhttp.config.IRequestApi;

/**
 * @author open_9527
 * Create at 2021/1/14
 **/
public class ShareApi implements IRequestApi {
    @Override
    public String getApi() {
        //user_article/list/0/json
        return "user_article/list/" + page + "/json";
    }
    @HttpIgnore
    private String page;

    public ShareApi setPage(int page) {
        this.page = String.valueOf(page);
        return this;
    }

}

File: feature/wanandroid/pkg/src/main/java/com/open9527/wanandroid/pkg/net/share/ShareApi.kt
package com.open9527.wanandroid.pkg.net.share

import com.android.open9527.okhttp.annotation.HttpIgnore
import com.android.open9527.okhttp.config.IRequestApi

/**
 * @author open_9527
 * Create at 2021/1/14
 */
class ShareApi : IRequestApi {
    override fun getApi(): String {
        //user_article/list/0/json
        return "user_article/list/$page/json"
    }

    @HttpIgnore
    private var page: String? = null
    fun setPage(page: Int): ShareApi {
        this.page = page.toString()
        return this
    }
}